lib/graphql/tracing/data_dog_tracing.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module GraphQL
+  module Tracing
+    class DataDogTracing < PlatformTracing
+      self.platform_keys = {
+        'lex' => 'lex.graphql',
+        'parse' => 'parse.graphql',
+        'validate' => 'validate.graphql',
+        'analyze_query' => 'analyze.graphql',
+        'analyze_multiplex' => 'analyze.graphql',
+        'execute_multiplex' => 'execute.graphql',
+        'execute_query' => 'execute.graphql',
+        'execute_query_lazy' => 'execute.graphql',
+      }
+
+      def platform_trace(platform_key, key, data)
+        pin = Datadog::Pin.get_from(self)
+        unless pin
+          pin = Datadog::Pin.new('graphql-ruby', app: 'graphql-ruby')
+          pin.onto(self)
+        end
+
+        pin.tracer.trace(platform_key, service: pin.service) do |span|
```

Comment:
For the service, it might be nice to conform to the same names that dd-trace-rb seems to uphold? So

"#{app-name}-#{service}". E.g. `myapp-controller`, `myapp-postgresql`.

It might be nice if this were configurable. E.g.

```
  use GraphQL::Tracing::DataDogTracing, service: "myapp-graphql"
```